opencxl/cxl/component/switch_connection_client.py

```diff
@@ -49,7 +49,7 @@ def __init__(
         if num_ld != 0:
             self._cxl_connection = [CxlConnection() for _ in range(num_ld)]
         else:
-            self._cxl_connection = CxlConnection()
+            self._cxl_connection = [CxlConnection()]
```

Comment:
What is the advantage of converting this to a list on non-MLD devices? i.e. what do we get in return for having to add `[0]` in quite a few places in the code?

Reply:
It is not a good practice to use the same variable for different types of contents (e.g., `List[CxlConnection]` and `CxlConnection`). Previously, the method `get_cxl_connection()` could return either a list of `CxlConnection`s (for MLDs) or just a single `CxlConnection` (for other cases), again, which is not a good practice. Therefore, I would like to unify the type of returned data into List[CxlConnection]` for conciseness.

Reply:
Got it. I think the right way to do this is to redesign MLD connections rather than conforming other devices to MLD's convention. I've had some reservations about that anyway.

If I am new to this code and look at:
```
        self._cxl_type1_device = CxlType1Device(
            CxlType1DeviceConfig(
                transport_connection=self._sw_conn_client.get_cxl_connection()[0],
```
It's not immediately obvious what [0] means and would wonder what [1...x] would mean.

Could you file a GitHub Issue to track "redesign MLD connections" and revert [0] code on this PR?